File: app/src/main/java/com/example/minicapstone390/Models/Sensor.java
package com.example.minicapstone390.Models;

import java.util.Map;

public class Sensor {
    private String id;
    private int SensorType;
    private String SensorName;
    private double SensorValue;
    private boolean status;
    private double SensorScore;

    public Sensor() {}

    public Sensor(String id, int sensorType, String sensorName, double sensorValue, boolean status, double sensorScore) {
        this.id = id;
        this.SensorType = sensorType;
        this.SensorName = sensorName;
        this.SensorValue = sensorValue;
        this.status = status;
        this.SensorScore = sensorScore;
    }

    public String getId() { return this.id; }

    public void setId(String id) { this.id = id; }

    public int getSensorType() { return this.SensorType; }

    public void setSensorType(int sensorType) { this.SensorType = sensorType; }

    public String getSensorName() { return this.SensorName; }

    public void setSensorName(String sensorName) { this.SensorName = sensorName; }

    public double getSensorValue() { return this.SensorValue; }

    public void setSensorValue(double sensorValue) { this.SensorValue = sensorValue; }

    public boolean getStatus() { return this.status; }

    public void setStatus(Boolean status) { this.status = status; }

    public double getSensorScore() { return this.SensorScore; }

    public void setSensorScore(double sensorScore) { this.SensorScore = sensorScore; }
}

File: Arduino/gasGuard/Sensor.cpp
#include "Sensor.h"
#include "Value.h"

Sensor::Sensor() {
  this.SensorValue = 0;
  this.SensorName = "";
  this.SensorType = 0;
  this.SensorPastValues = new map<String, Value>();
  this.status = false;
}

Sensor::Sensor(String SensorName, Value value, int SensorType, float SensorValue, ) {
  this.SensorValue = SensorValue;
  this.SensorName = SensorName;
  this.SensorType = SensorType;
  this.SensorPastValues.insert(value.getTimeStamp(), value.getValue(); // hmm not sure how to go about this... also this should probs just initialize an empty map do this add in the addPastValue
  this.status = true;
}


String Sensor::getSensorName() {
  return this.SensorName;
}
float Sensor::getSensorValue() {
  return this.SensorValue;
}
bool Sensor::getStatus() {
  return this.status;
}
map<String, Value> Sensor::getSensorPastValues() {
  return this.SensorPastValues;
}

void Sensor::setSensorName(String SensorName) {
  this.SensorName = SensorName
}
void Sensor::setSensorValue(float SensorValue) {
  this.SensorValue = SensorValue
}
void Sensor::setStatus(bool status) {
  this.status = status;
}
void Sensor::addPastValue(Value) {
  // do logic here to add....
}
